proto/github.proto

```diff
@@ -21,3 +21,170 @@ message UnlinkGitHubAccountResponse {
+// be used to re-assign the org that an installation is linked to, or to unlink
+// an installation from an org (note that unlinking is not the same as
+// uninstalling).
+
+// A request to link a GitHub App installation to the user's currently selected
+// BuildBuddy organization.
+//
+// The user must have a personal GitHub account linked to BuildBuddy, and that
+// account must have access to the installation ID.
+message LinkAppInstallationRequest {
+  context.RequestContext request_context = 1;
+
+  // The installation ID. This is only set after completing the first-time
+  // setup. If the app is already installed, this can be blank. The server will
+  // return a NotFound error if this is blank and the app is not yet installed.
+  int64 installation_id = 2;
+}
+
+message LinkAppInstallationResponse {
+  context.ResponseContext response_context = 1;
+}
+
+message AppInstallation {
+  // The GitHub App installation ID.
+  int64 installation_id = 1;
+
+  // The GitHub username or organization where the app is installed.
+  // This will be unique for each installation.
+  string owner = 2;
+
+  // The group ID to which the installation is linked.
+  string group_id = 3;
+}
+
+// A request to list GitHub App installations accessible to user as well as
+// installations linked to the org (which may or may not be directly accessible
+// to the user).
+//
+// Installations are returned for all groups that are accessible to the user, so
+// that the user can get a clear view of which installations are linked to which
+// orgs, without needing to switch orgs.
+message GetAppInstallationsRequest {
+  context.RequestContext request_context = 1;
+}
+
+message GetAppInstallationsResponse {
+  context.ResponseContext response_context = 1;
+
+  // The installations available to the user.
+  repeated AppInstallation installations = 2;
+}
+
+// A request to update a GitHub App installation. The authenticated user must
+// have a linked GitHub account which has access to the installation.
+message UpdateAppInstallationRequest {
+  context.RequestContext request_context = 1;
+
+  // The installation ID to update.
+  // Required.
+  int64 installation_id = 2;
+
+  // The group to assign or re-assign the installation to.
+  // Required.
+  string group_id = 3;
+}
+
+message UpdateAppInstallationResponse {
+  context.ResponseContext response_context = 1;
+}
+
+// A request to unlink a GitHub App installation from the authenticated org.
+//
+// The authenticated user does not have to have access to the installation in
+// order to unlink it from the org.
+//
+// Unlinking an installation does not automatically unlink all repositories
+// associated with the installation. However, members of the organization will
+// no longer be able to get repository access tokens (unless the installation is
+// re-linked).
+message UnlinkAppInstallationRequest {
+  context.RequestContext request_context = 1;
+
+  // The installation ID to unlink.
+  int64 installation_id = 2;
```

Comment:
if `UpdateAppInstallationRequest` needs `group_id` i assume this needs one, too? Or does `Update` not actually need it?

Reply:
So re. `Update` having a group ID, my original thinking was that in the settings page, we can allow the user to see all the GitHub installations they have access to, as well as which organizations they are linked to (if any). And if the installation is linked to a different org, the user can use this RPC to re-assign the the installation to a different org.

But it probably won't be super common that users want to re-assign an installation to a different org, so I've removed the group_id field from `UpdateAppInstallationRequest`. Instead of re-assigning, they can always Unlink then re-link (as a different group).

re. `Unlink` not needing a group_id, we don't need a group ID because installations can only be assigned to one group, so the group_id comes from the GitHubAppInstallation row in the DB.

Reply:
Getting rid of updated makes sense to me.  One remaining wrinkle is that this means an "unlink" operation will blow away the linkage even if someone else has made a change to the linkage in the meantime.  This probably doesn't matter, but the group ID could be useful here just to confirm that you're operating on the state that you meant to operate on.